### PR TITLE
Add StatCalculator Class and Initial Tests for Statistics Calculations

### PR DESCRIPTION
**Pull request by qoryhanisagal**
#### Summary

This pull request introduces the `StatCalculator` class, designed to handle core statistical calculations required across different parts of the project. It also includes an initial set of tests to confirm the class’s setup and ensure accurate calculations.

#### Details

- **Class Added**: `StatCalculator`
  - **Purpose**: Centralizes statistical calculations to adhere to the DRY and SRP principles, avoiding code repetition across multiple files.
- **Methods**: No methods implemented at this time.
- **Testing**:
  - Created `stat_calculator_spec.rb` to validate the existence and functionality of the `StatCalculator` class.
  - **Example Tests**: Includes an initial test to verify that `StatCalculator` exists and can be instantiated.

#### Author Attribution

- Authored by **Sequoyah Descoteaux** (`sequoyah.m.dozier@gmail.com`), with initial work and setup.

#### Next Steps

- As we implement further methods within `StatCalculator`, additional tests will be added to `stat_calculator_spec.rb` to confirm functionality and accuracy.

### CODE REVIEW

File: lib/stat_calculator.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# lib/stat_calculator.rb
```
Comment (DustinPeukert) [inline lib/stat_calculator.rb:1]: What is this comment for?

**Comment by DustinPeukert** [comment]
Make sure to have a commit message for every commit. It seems one was forgotten this time.

**Comment by KMPWeiland** [comment]
Looks good! Looking forward to adding all the methods here.

File: spec/stat_calculator_spec.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# spec/stat_calculator_spec.rb
+require 'rspec'
+require_relative '../lib/stat_calculator'
```
Comment (DustinPeukert) [inline spec/stat_calculator_spec.rb:3]: no need to make individual requires here. We have a spec_helper file that you can add to:
just throw "require './lib/stat_calculator'" in the spec_helper file
then put "require_relative 'spec_helper'" in this file

Note for the future: require statements are for general use as well as all files that are not in the immediate directory
require_relative is only every to be used when requiring a file that is within the same directory

**Review by DustinPeukert — APPROVED** [review]

File: spec/stat_calculator_spec.rb (hunk 2 above)
Comment (qoryhanisagal) [reply spec/stat_calculator_spec.rb:3]: Thanks for the feedback however it was added as pre-cursors for our next step. However, I installed rspec on the branch for tomorrow session with add method to the stat_calculator. I will keep the require 'rspec'. 

**Changes made based on your review:**

- Added require './lib/stat_calculator' to the spec_helper file
- Added the 'require_relative spec_helper' in the spec/stat_calculator_spec.rb file

**Review by kevin-newland — APPROVED** [review]
Everything looks great!!

File: lib/stat_calculator.rb (hunk 1 above)
Comment (qoryhanisagal) [reply lib/stat_calculator.rb:1]: when files are created in my VS code it adds that comment to the files

It's with the file auto-generated documentation that relies on header comments to understand each file’s role.

I did however add this "# This file contains the StatCalculator class, responsible for performing core statistical calculations across the project."

**Pull request merged**